src/main/java/com/guciowons/shoppingbasket/Basket/Basket.java

```diff
@@ -1,13 +1,21 @@
 package com.guciowons.shoppingbasket.Basket;
 
 import com.guciowons.shoppingbasket.Exception.NoProductInBasketException;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.annotation.Version;
+import org.springframework.data.mongodb.core.mapping.Document;
 
 import java.util.HashMap;
 
+@Document
 public class Basket {
-    private final int id;
+    @Id
+    private int id;
```

Comment:
```
In MongoDB, the ID is composed of 12 bytes:

a 4-byte timestamp that represents ID's creation measured in seconds since the Unix epoch
a 5-byte random generated value that is unique to the machine and process
a 3-byte incrementing counter
```

Więc w MongoDB to musi być String

Reply:
Chyba, że to działa tak jak teraz jest to okej. Ale raczej używa się Stringa przy mongo

Reply:
Poza tym widzę, że trochę źle tu używasz tej adnotacji. `@Id` oznacza, że twoim ID będzie zarządzał system bazodanowy, więc będzie je też nadawał. A Ty tutaj sam wpisujesz to ID przy tworzeniu. Popraw to, żeby było zgodne ze standardami.